include/openthread/platform/radio.h

```diff
@@ -108,6 +108,12 @@ typedef struct otRadioFrame
     bool     mIsARetx : 1;       ///< Set to true if this frame is a retransmission. Should be ignored by radio driver.
     bool     mIsCcaEnabled : 1;  ///< Set to true if CCA must be enabled for this packet. False otherwise.
 
+    // Applicable/Required only when time synchronization service (`OPENTHREAD_CONFIG_ENABLE_TIME_SYNC`) is enabled.
```

Comment:
How about if instead of having these be part of the `otRadioFrame` struct, we add new APIs to set and get these parameters . Probably one API to set the parameters for the next frame to be sent, and one to get the values for the current/last frame received?

Reply:
Does it aim to save Ram space when these parameters are not used? If so, how about only add a new pointer here which points to a new `HeaderIeInfo` structure? It will be initialized as NULL if Header IE related features are not enabled.